src/reference_impls/memory.rs

```diff
@@ -552,14 +553,29 @@ impl Memory for SimpleMemory {
 
     fn specialized_code_query(
         &mut self,
-        _monotonic_cycle_counter: u32,
-        query: MemoryQuery,
+        monotonic_cycle_counter: u32,
+        mut query: MemoryQuery,
     ) -> MemoryQuery {
         assert_eq!(query.location.memory_type, MemoryType::Code);
         let page = query.location.page.0;
 
+        // There are two types of places where decommitment could happen:
```

Comment:
Actually you do not need to check heap pages here unless we have a wrong implementation somewhere above. EVM simulator can only access opcode via uma.fat_ptr_read and this function is used only for VM opcodes

Reply:
but decommitter actually uses it to write to the memory of the code oracle